js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
```

Comment:
I don't see it as "tech debt". I thought about this API carefully. While it lacks some ergonomic niceties, it has other desirable properties (statelessness, testability, self-documenting-ness, explicitness)

Using a `Validator` object that stores the style internally, we would either have to 

 - implement style mutation methods internally (e.g. `Validator#addLayer`) or
 - re-create the `Validator` every time the style changes (which is after most calls to validate) 

I'm not against refactoring towards a more ergonomic design, but I think it can be a separate lift.